python/ray/_private/utils.py

```diff
@@ -1884,18 +1884,8 @@ def update_envs(env_vars: Dict[str, str]):
     if not env_vars:
         return
 
-    replaceable_keys = [
-        "PATH",
-        "LD_LIBRARY_PATH",
-        "DYLD_LIBRARY_PATH",
-        "LD_PRELOAD",
-    ]
-
     for key, value in env_vars.items():
-        if key in replaceable_keys:
-            os.environ[key] = value.replace("${" + key + "}", os.environ.get(key, ""))
-        else:
-            os.environ[key] = value
+        os.environ[key] = os.path.expandvars(value)
```

Comment:
Looks like the Python implementation leaves the env var unchanged if it doesn't exist. E.g.,

```
unset ENV_VAR

# previous
${ENV_VAR} -> ""

# now
${ENV_VAR} -> ${ENV_VAR}
```

I feel like this should be fine (especially given this python implementation will work cross platform correctly). But if we have to keep this behavior, I can somehow write the custom implementation

Reply:
nice, using python's expandvars is fine